app/actions.tsx

```
@@ -644,10 +751,19 @@ export const AI = createAI<AIState, UIState>({
 export const getUIStateFromAIState = (aiState: AIState): UIState => {
   const chatId = aiState.chatId
   const isSharePage = aiState.isSharePage
+
+  // Filter messages to only include the last 'data' message if multiple exist
+  const lastDataMessageIndex = [...aiState.messages].reverse().findIndex(m => m.role === 'data')
+  const actualLastDataIndex = lastDataMessageIndex === -1 ? -1 : aiState.messages.length - 1 - lastDataMessageIndex
+
   return aiState.messages
     .map((message, index) => {
```

Biome (2.3.14)

[error] This callback passed to map() iterable method should always return a value.

Add missing return statements so that this callback returns a value on all execution paths.

(lint/suspicious/useIterableCallbackReturn)